resources/monitor_activity_logs.go

```diff
@@ -232,8 +234,13 @@ func MonitorActivityLogs() *schema.Table {
 //                                               Table Resolver Functions
 // ====================================================================================================================
 func fetchMonitorActivityLogs(ctx context.Context, meta schema.ClientMeta, parent *schema.Resource, res chan interface{}) error {
+	// we fetch activity logs records from now to some point in the past. fetchWindow defines how far that point in the past is.
+	const fetchWindow = 90 * 24 * time.Hour
 	svc := meta.(*client.Client).Services().Monitor.ActivityLogs
-	response, err := svc.List(ctx, "", "")
+	now := time.Now().UTC()
+	past := now.Add(-fetchWindow)
+	filter := fmt.Sprintf("eventTimestamp ge '%s' and eventTimestamp le '%s'", past.Format(time.RFC3339Nano), now.Format(time.RFC3339Nano))
+	response, err := svc.List(ctx, filter, "")
```

Comment:
We might want to put a default limit for last 24h otherwise this fetch could take longer and longer imo

Reply:
90 days window is required by the Azure CIS policy.

Reply:
:+1: I would limit it to 90 days then.

Reply:
Isn't it limited to 90 days? Or am I misunderstanding your comment somehow?